great_expectations/core/batch_definition.py

```diff
@@ -76,13 +76,17 @@ def get_batch(self, batch_parameters: Optional[BatchParameters] = None) -> Batch
 
         return batch_list[-1]
 
-    def is_added(self) -> tuple[bool, list[ResourceNotAddedError]]:
+    def is_added(self) -> tuple[bool, ResourceNotAddedError | None]:
```

Comment:
Deciding to make leaf nodes just return a single instance instance of a list to make it easier to raise (alternative would have been `raise errors[0]` and that feels bad)